### PR TITLE
[chore] Change exporter config tests to unmarshal config only for that component. (part1) 

### PR DESCRIPTION
**Pull request by kovrus**
**Description:** <Describe what has changed.>

Change exporters config tests to unmarshal config only for that component.

- [x] alibabacloudlogserviceexporter
- [x] awscloudwatchlogsexporter
- [x] awsemfexporter
- [x] awskinesisexporter
- [x] awsxrayexporter
- [x] azuredataexplorerexporter
- [x] azuremonitorexporter
- [x] carbonexporter
- [x] clickhouseexporter
- [x] coralogixexporter
- [x] datadogexporter
- [x] dynatraceexporter
- [x] elasticsearchexporter

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13224

### CODE REVIEW

**Comment by kovrus** [comment]
cc: @bogdandrutu

**Comment by kovrus** [comment]
@dmitryax can you also look at this PR? It is similar to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14954

File: exporter/coralogixexporter/testdata/config-all.yaml [hunk 1]
```diff
@@ -6,22 +6,7 @@ processors:
...
-      endpoint: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
-    application_name_attributes:
-    - "service.namespace"
-    subsystem_name_attributes:
-    - "service.name"
-    private_key: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
-    application_name: "APP_NAME"
-    subsystem_name: "SUBSYSTEM_NAME"
-    timeout: 5s
+
```
Comment (dmitryax) [inline exporter/coralogixexporter/testdata/config-all.yaml:9]: This file should be removed?

Comment (kovrus) [reply exporter/coralogixexporter/testdata/config-all.yaml]: done

**Review by dmitryax — APPROVED** [review]

File: exporter/awscloudwatchlogsexporter/config_test.go [hunk 2]
```diff
@@ -20,84 +20,87 @@ import (
...
+			id:           config.NewComponentIDWithName(typeStr, "invalid_queue_size"),
+			errorMessage: "'sending_queue.queue_size' must be 1 or greater",
+		},
+		{
+			id:           config.NewComponentIDWithName(typeStr, "invalid_required_field_stream"),
+			errorMessage: "'log_stream_name' must be set",
+		},
+		{
+			id:           config.NewComponentIDWithName(typeStr, "invalid_required_field_group"),
+			errorMessage: "'log_group_name' must be set",
```
Comment (codeboten) [inline exporter/awscloudwatchlogsexporter/config_test.go:85]: it looks like a test for `invalid_queue_setting` is missing here

**Review by codeboten — COMMENTED** [review]
Just one comment, thanks for cleaning up these tests!

File: exporter/awscloudwatchlogsexporter/config_test.go (hunk 2 above)
Comment (kovrus) [reply exporter/awscloudwatchlogsexporter/config_test.go:86]: Added it back

**Pull request merged**